src/test/java/com/exsol/errorcodemodel/ErrorCodeParserTest.java

```diff
@@ -4,34 +4,48 @@
 import static org.hamcrest.Matchers.equalTo;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import com.exsol.errorcodemodel.ErrorIdentifier.SyntaxException;
+import com.exsol.errorcodemodel.ErrorIdentifier.Type;
 
 class ErrorCodeParserTest {
     private static final ErrorCodeParser ERROR_CODE_READER = new ErrorCodeParser();
 
-    @Test
-    void testReadValidCode() throws ErrorIdentifier.SyntaxException {
-        assertThat(ERROR_CODE_READER.parse("E-EXA-1"), equalTo(new ErrorIdentifier(ErrorIdentifier.Type.E, "EXA", 1)));
-    }
-
-    @Test
-    void testReadValidCodeWithSubTag() throws ErrorIdentifier.SyntaxException {
-        assertThat(ERROR_CODE_READER.parse("F-EXA-E1-1"),
-                equalTo(new ErrorIdentifier(ErrorIdentifier.Type.F, "EXA-E1", 1)));
+    @ParameterizedTest
+    @CsvSource(value = { //
+            "E-EXA-1, E, EXA, 1", //
```

Comment:
Test `SQL-1234` also. This is the main error code style used in the core.

Reply:
✅